src/legacy/app/actions/deviceActions.js

```diff
@@ -224,52 +143,76 @@ export function deselectDevice() {
     };
 }
 
+export const wrapDevice = device => {
+    const outputDevice = camelcaseKeys(device, { deep: true });
+    const serialport = outputDevice.serialPorts
+        ? outputDevice.serialPorts[0]
+        : undefined;
+    return {
+        ...outputDevice,
+        boardVersion: outputDevice.jlink
+            ? outputDevice.jlink.boardVersion
+            : undefined,
+        serialport,
+    };
+};
+
+export const wrapDevices = devices => devices.map(wrapDevice);
+
 /**
  * Starts watching for devices with the given traits. See the nrf-device-lister
  * library for available traits. Whenever devices are attached/detached, this
  * will dispatch DEVICES_DETECTED with a complete list of attached devices.
  *
  * @returns {function(*)} Function that can be passed to redux dispatch.
  */
-export function startWatchingDevices() {
-    const config = getAppConfig();
+export const startWatchingDevices = () => async (dispatch, getState) => {
+    const updateDeviceList = async () => {
+        const devices = wrapDevices(
+            await nrfDeviceLib.enumerate(
+                deviceLibContext,
+                getAppConfig().selectorTraits
+            )
+        );
+        const { device } = getState();
 
-    return (dispatch, getState) => {
-        if (!deviceLister) {
-            deviceLister = new DeviceLister(config.selectorTraits);
+        if (
+            device &&
```

Comment:
```suggestion
            device?.selectedSerialNumber !== null &&
```

Reply:
Hmm, if we make that change we have to update the comparison to `device?.selectedSerialNumber != null` right? Otherwise `device?` might be `undefined` which does not equal `null` and the conditional would wrongly evaluate to true. So we need to check for both `null` and `undefined`.

Reply:
Good point :)